File: src/B_10707/B_10707.java
package B_10707;

import java.io.*;

public class B_10707 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int a = Integer.parseInt(br.readLine());
        int b = Integer.parseInt(br.readLine());
        int c = Integer.parseInt(br.readLine());
        int d = Integer.parseInt(br.readLine());
        int p = Integer.parseInt(br.readLine());
        System.out.println(Math.min(xCom(a,p),yCom(b,c,d,p)));

    }

    static int xCom(int a, int p){
        return a*p;
    }

    static int yCom(int b, int c, int d,int p){
        if(p>c){
            return b + ((p-c)*d);
        }else{
            return b;
        }
    }
}

File: src/B_10707/B_10707.kt
package B_10707

import java.io.*

fun main() = with(BufferedReader(InputStreamReader(System.`in`))){
    val a: Int = readLine().toInt()
    val b: Int = readLine().toInt()
    val c: Int = readLine().toInt()
    val d: Int = readLine().toInt()
    val p: Int = readLine().toInt()
    println(xCom(a, p).coerceAtMost(yCom(b, c, d, p)))
}
fun xCom(a: Int, p: Int): Int {
    return a * p
}

fun yCom(b: Int, c: Int, d: Int, p: Int): Int {
    return if (p > c) {
        b + (p - c) * d
    } else {
        b
    }
}